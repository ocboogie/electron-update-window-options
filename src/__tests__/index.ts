import electronUpdateWindowOptions from "../";

it("changes browser window", () => {
  const mock = {
    setAlwaysOnTop: jest.fn()
  };
  electronUpdateWindowOptions(mock as any, { alwaysOnTop: true });
  expect(mock.setAlwaysOnTop.mock.calls[0]).toEqual([true]);
});

describe("multiple argument options", () => {
  it("changes browser window", () => {
    const mock = {
      setPosition: jest.fn()
    };
    electronUpdateWindowOptions(mock as any, { x: 4, y: 2 });
    expect(mock.setPosition.mock.calls[0]).toEqual([4, 2]);
  });

  it("throws when not paired", () => {
    const filler = {
      setPosition: () => undefined
    };
    expect(() => {
      electronUpdateWindowOptions(filler as any, { y: 4 });
    }).toThrowErrorMatchingSnapshot();
  });
});
