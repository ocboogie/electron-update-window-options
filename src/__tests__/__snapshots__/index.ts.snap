// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple argument options throws when not paired 1`] = `"\\"y\\" is missing its pair(s). You must have \\"x\\",\\"y\\""`;
